bqskit/ir/lang/qasm2/visitor.py

```diff
@@ -650,9 +646,77 @@ def measure(self, tree: lark.Tree) -> None:
                 'measured to a single classical bit.',
             )
 
+        params: list[float] = []
+        qlist = tree.children[0]
+        location = CircuitLocation(self.convert_qubit_ids_to_indices(qlist))
+
+        # Parse gate object
+        gate_name = tree.data
+        if gate_name in self.gate_defs:
+            gate_def: GateDef | CustomGateDef = self.gate_defs[gate_name]
+        elif gate_name in self.custom_gate_defs:
+            gate_def = self.custom_gate_defs[gate_name]
+        else:
+            raise LangException('Unrecognized gate: %s.' % gate_name)
+
+        if len(params) != gate_def.num_params:
+            raise LangException(
+                'Expected %d params got %d params for gate %s.'
+                % (gate_def.num_params, len(params), gate_name),
+            )
+
+        if len(location) != gate_def.num_vars:
+            raise LangException(
+                'Gate acts on %d qubits, got %d qubit variables.'
+                % (gate_def.num_vars, len(location)),
+            )
+
+        # Build operation and add to circuit
+        self.op_list.append(gate_def.build_op(location, params))
```

Comment:
I don't understand the point of this block of code. In the if-elif-else block above we checked if it was either a `measure qreg -> creg` or a `measure qubit -> clbit` or invalid. Those branches, in your code, conclude with adding something to the gate_def dictionary. They should not change the gate_def dictionary, instead they should conclude with adding something to the `op_list` directly.